cli/bridge-admin/bridging_addresses_balances_params.go

```diff
@@ -0,0 +1,199 @@
+package clibridgeadmin
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/Ethernal-Tech/apex-bridge/common"
+	ethtxhelper "github.com/Ethernal-Tech/apex-bridge/eth/txhelper"
+	vcCore "github.com/Ethernal-Tech/apex-bridge/validatorcomponents/core"
+	"github.com/Ethernal-Tech/cardano-infrastructure/indexer"
+	indexerDb "github.com/Ethernal-Tech/cardano-infrastructure/indexer/db"
+	cardanowallet "github.com/Ethernal-Tech/cardano-infrastructure/wallet"
+	"github.com/spf13/cobra"
+)
+
+const (
+	primeWalletAddressFlag  = "prime-wallet-addr"
+	vectorWalletAddressFlag = "vector-wallet-addr"
+	nexusWalletAddressFlag  = "nexus-wallet-addr"
+	indexerDbsPathFlag      = "indexer-dbs-path"
+
+	primeWalletAddressFlagDesc  = "prime wallet address"
+	vectorWalletAddressFlagDesc = "vector wallet address"
+	nexusWalletAddressFlagDesc  = "nexus wallet address"
+	indexerDbsPathFlagDesc      = "path to the indexer database"
+)
+
+type bridgingAddressesBalancesParams struct {
+	config              string
+	primeWalletAddress  string
+	vectorWalletAddress string
+	nexusWalletAddress  string
+	indexerDbsPath      string
+}
+
+func (b *bridgingAddressesBalancesParams) ValidateFlags() error {
+	if b.primeWalletAddress == "" || b.vectorWalletAddress == "" || b.nexusWalletAddress == "" {
+		return fmt.Errorf("all wallet addresses --%s, --%s and --%s must be set",
+			primeWalletAddressFlag, vectorWalletAddressFlag, nexusWalletAddressFlag)
+	}
+
+	if !common.IsValidAddress(common.ChainIDStrPrime, b.primeWalletAddress) {
+		return fmt.Errorf("invalid address: --%s", primeWalletAddressFlag)
+	}
+
+	if !common.IsValidAddress(common.ChainIDStrVector, b.vectorWalletAddress) {
+		return fmt.Errorf("invalid address: --%s", vectorWalletAddressFlag)
+	}
+
+	if !common.IsValidAddress(common.ChainIDStrNexus, b.nexusWalletAddress) {
+		return fmt.Errorf("invalid address: --%s", nexusWalletAddressFlag)
+	}
+
+	if b.config == "" {
+		return fmt.Errorf("invalid config path: --%s", indexerDbsPathFlag)
+	}
+
+	return nil
+}
+
+func (b *bridgingAddressesBalancesParams) RegisterFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(
+		&b.config,
+		configFlag,
+		"",
+		configFlagDesc,
+	)
+	cmd.Flags().StringVar(
+		&b.primeWalletAddress,
+		primeWalletAddressFlag,
+		"",
+		primeWalletAddressFlagDesc,
+	)
+	cmd.Flags().StringVar(
+		&b.vectorWalletAddress,
+		vectorWalletAddressFlag,
+		"",
+		vectorWalletAddressFlagDesc,
+	)
+	cmd.Flags().StringVar(
+		&b.nexusWalletAddress,
+		nexusWalletAddressFlag,
+		"",
+		nexusWalletAddressFlagDesc,
+	)
+	cmd.Flags().StringVar(
+		&b.indexerDbsPath,
+		indexerDbsPathFlag,
+		"",
+		indexerDbsPathFlagDesc,
+	)
+}
+
+func (b *bridgingAddressesBalancesParams) Execute(outputter common.OutputFormatter) (common.ICommandResult, error) {
+	appConfig, err := common.LoadConfig[vcCore.AppConfig](b.config, "")
+	if err != nil {
+		return nil, err
+	}
+
+	if b.indexerDbsPath != "" { // Retrieve Cardano balances from the database
```

Comment:
can we retrieve this db path from config?
maybe it would be easier to have bool flag for command (use idexer database or use ogmios)

Reply:
After discussion with Zoltan we agree to leave this flag as it is